engine/classes/Elgg/Filesystem/Adapter/AwsS3.php

```diff
@@ -0,0 +1,48 @@
+<?php
+namespace Elgg\Filesystem\Adapter;
+use Gaufrette\Filesystem;
+use Gaufrette\File;
+use Gaufrette\Adapter\AwsS3 as GaufretteAwsS3;
+use Aws\S3\S3Client;
+
+class AwsS3 extends Filesystem implements Adapter {
```

Comment:
Yes, I meant to inherit from Filesystem to avoid either re-implementing or wrapping the entire Gaufrette class.